lib/cloudflare_protected_email_address.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'cgi'
+
+class CloudflareProtectedEmailAddress
+  def initialize(obfuscated)
+    @obfuscated = obfuscated
+  end
+
+  def human_readable
+    return CGI.unescape(escaped_email_address) if CGI.unescape(escaped_email_address).include?('@')
+    obfuscated
```

Comment:
The logic of this also seems back-to-front. A guard clause should be used to break out in unusual cases, not be the main path, with a fall through to the odd ones.

Again, I'm not convinced you need this at all, but if you do, then it would be better as:

```ruby
  return obfuscated unless unescaped.include? '@'
  unescaped
```